submissions/kazeemErinfolami/javascript/Dom/index.js

```diff
@@ -0,0 +1,76 @@
+let qty = document.getElementById('qty'),
+    price = document.getElementById('price'),
+    shippingCost = document.getElementById('shippingCost'),
+    cost = document.getElementById('total'),
+    button = document.getElementById('buyButton');
+
+
+//initial value
+let priceValue = price.innerHTML,
+    costValue = cost.innerHTML;
+
+//RegEx
+const Regex = /^[1-9]/;
+
+function productQuantity() {
+    if (!qty.value.match(Regex)) {
+        button.disable = true
+        button.style = "background-color: grey"
+        qty.nextElementSibling.innerHTML = "⚠️Please enter a valid quantity";
+    } else {
+        button.disabled = false;
+        qty.nextElementSibling.innerHTML = "";
+    }
+}
+function productPrice() {
+    if (qty.value >= 1) {
+        let perPrice = priceValue * qty.value
+        price.innerHTML = perPrice
+    } else if (qty.value === "") {
+        price.innerHTML = priceValue
+        cost.innerHTML = costValue
+    }
+    else {
+        return false
+    }
+}
+
+function shippingPrice() {
+    let perPrice = priceValue * qty.value
+    if (perPrice < 1000) {
+        shippingRange = 100;
+        shippingCost.innerHTML = shippingRange;
+    } else if ((perPrice >= 1000) && (perPrice <= 20000)) {
+        shippingRange = 0.15 * perPrice;
+        shippingCost.innerHTML = shippingRange;
+    } else if ((perPrice >= 20000) && (perPrice <= 100000)) {
+        shippingRange = 0.2 * perPrice;
+        shippingCost.innerHTML = shippingRange;
+    } else if (perPrice >= 100000) {
+        shippingRange = 25000;
+        shippingCost.innerHTML = shippingRange;
+    } else {
+        return false
+    }
+}
+
+function totalPrice() {
+    let perPrice = priceValue * qty.value
+    if (qty.value >= 1) {
+        cost.innerHTML = shippingRange + perPrice;
```

Comment:
How come you are making use of `shippingRange` here. Where is the value coming from?

Reply:
This function should just hold this `cost.innerHTML = shippingRange + perPrice;`. `shippingRange` and `perPrice` should be a parameter and the function should be called inside the `shippingPrice` function as `shippingRange` and `perPrice` is already calculated there

Reply:
I started making use of the shippingRange in the "function shippingPrice()" block, I should have declared it as a variable, my bad.